modules/komoonaBidAdapter.js

```diff
@@ -1,117 +1,123 @@
-  baseAdapter.callBids = function(bidRequest) {
-    const bids = bidRequest.bids || [];
-    const tags = bids
-      .filter(bid => valid(bid))
-      .map(bid => {
-        // map request id to bid object to retrieve adUnit code in callback
-        bidRequests[bid.bidId] = bid;
-
-        let tag = {};
-        tag.sizes = bid.sizes;
-        tag.uuid = bid.bidId;
-        tag.placementid = bid.params.placementId;
-        tag.hbid = bid.params.hbid;
+      return tag;
+    });
 
-        return tag;
-      });
+    // Komoona server config
+    const time = new Date().getTime();
+    const kbConf = {
+      ts_as: time,
+      hb_placements: [],
+      hb_placement_bidids: {},
+      hb_floors: {},
+      cb: _generateCb(time),
+      tz: new Date().getTimezoneOffset(),
+    };
+
+    validBidRequests.forEach(bid => {
+      kbConf.hdbdid = kbConf.hdbdid || bid.params.hbid;
+      kbConf.encode_bid = kbConf.encode_bid || bid.params.encode_bid;
+      kbConf.hb_placement_bidids[bid.params.placementId] = bid.bidId;
+      if (bid.params.floorPrice) {
+        kbConf.hb_floors[bid.params.placementId] = bid.params.floorPrice;
+      }
+      kbConf.hb_placements.push(bid.params.placementId);
+    });
 
+    let payload = {};
     if (!utils.isEmpty(tags)) {
-      const payload = JSON.stringify({bids: [...tags]});
-
-      ajax(ENDPOINT, handleResponse, payload, {
-        contentType: 'text/plain',
-        withCredentials: true
-      });
+      payload = { bids: [...tags], kbConf: kbConf };
     }
-  };
-
-  /* Notify Prebid of bid responses so bids can get in the auction */
-  function handleResponse(response) {
-    let parsed;
 
+    return {
+      method: 'POST',
+      url: ENDPOINT,
+      data: JSON.stringify(payload)
+    };
+  },
+  /**
+  * Unpack the response from the server into a list of bids.
+  *
+  * @param {*} response A successful response from the server.
+  * @return {Bid[]} An array of bids which were nested inside the server.
+  */
+  interpretResponse: (response, request) => {
+    const bidResponses = [];
     try {
-      parsed = JSON.parse(response);
-    } catch (error) {
-      utils.logError(error);
-    }
-
-    if (!parsed || parsed.error) {
-      let errorMessage = `in response for ${baseAdapter.getBidderCode()} adapter`;
-      if (parsed && parsed.error) { errorMessage += `: ${parsed.error}`; }
-      utils.logError(errorMessage);
-
-      // signal this response is complete
-      Object.keys(bidRequests)
-        .map(bidId => bidRequests[bidId].placementCode)
-        .forEach(placementCode => {
-          bidmanager.addBidResponse(placementCode, createBid(STATUS.NO_BID));
+      if (response.bids) {
+        response.bids.forEach(bid => {
+          // The bid ID. Used to tie this bid back to the request.
+          bid.requestId = bid.uuid;
+          // The creative payload of the returned bid.
+          bid.ad = bid.creative;
+          bidResponses.push(bid);
```

Comment:
There are lots of required params missing from these bids. See: https://github.com/prebid/Prebid.js/pull/1738/files#diff-ac0af2601dcafed83fa9a11b814bf3b5L109

Reply:
Our md file was updated with correct placement Id and hb Id for test creative (as well as our servers) and now when using placementId and hbId from md file, the hello_world file will always return a creative

Reply:
Working to fix it, it was added to our servers and now testing it out - will ping once live and fixed